code/modules/projectiles/guns/energy/lawgiver.dm

```diff
@@ -188,5 +213,7 @@
 		message_enabled = 1
 		message_disable = 0
 		play_message()
-	else
-		return
+	
+	var/datum/firemode/new_mode = firemodes[sel_mode]
+	new_mode.apply_to(src)
```

Comment:
That would get and apply the firemode every time the owner (or anyone else, if emagged) speaks near the lawgiver.

I would put that part into the different if clauses, where the firemodes are changed.
It adds a bit of duplication but avoids calling that proc when not needed.